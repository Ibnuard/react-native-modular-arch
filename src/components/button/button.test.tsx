import renderer from 'react-test-renderer';
import {fireEvent, render, screen} from '@testing-library/react-native';
import Button from '.';
// test button render correctly
describe('button render tests', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Button title="Button" />);
    expect(tree).toMatchSnapshot();
  });

  it('should render given title', () => {
    const title = 'Button';
    const {getByText} = render(<Button title={title} />);
    const titleElement = getByText(title);

    expect(titleElement).toBeDefined();
  });

  it('calls onPress when pressed', () => {
    const onPressMock = jest.fn();
    const {getByText} = render(
      <Button title="Test Button" onPress={onPressMock} />,
    );
    const button = getByText('Test Button');

    fireEvent.press(button);

    expect(onPressMock).toHaveBeenCalled();
  });
});
