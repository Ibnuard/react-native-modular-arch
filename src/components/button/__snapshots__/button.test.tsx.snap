// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button render tests should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffa418",
      "borderRadius": 8,
      "justifyContent": "center",
      "opacity": 1,
      "paddingVertical": 23.684210526315788,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "Poppins-Regular",
        "fontSize": 24,
        "fontWeight": "normal",
      }
    }
  >
    Button
  </Text>
</View>
`;
